### ISSUE TITLE
gatsby develop fails

### ISSUE THREAD

**Issue by sparker888** [bagseye/barcadia#4]
Your site looks fantastic. I'm still learning Gatsby, but I've created two sites now using Contentful, so I know the basics. So, I have my space id and token set up properly. The develop fails with  TypeError: Cannot read property 'products' of undefined and then multiple GraphQL errors as it appears to be looking for both Products and Features data. Does the data need to be populated into Contentful first for this to work? Thank you.

**Comment by bagseye** [comment]
Yes, I think it may require content, even dummy content for it to build. I'll probably need to to build in some sort of check the prevent future build problems but for the moment can you make some dummy content to see if that solves the errors

**Comment by sparker888** [comment]
I don't have the time this week, but I will look at it again once I have another project rolling. It would be nice to get it working!

**Comment by chengjie0316** [comment]
I am facing the same problem , hey sparker had you solved it ??

**Comment by bagseye** [comment]
If it is the error previously reported, you will need to have some 'dummy content' in Contentful for it to build. There is a fix here -  https://github.com/gatsbyjs/gatsby/discussions/32009#discussioncomment-1063860 - to define types, but I don't have time to implement it right now.

**Comment by ibragimof** [comment]
I'm getting "product" pages and its widgets without pictures. Also all blog post coming with no picture from Contentful.
While `gatsby develop` got this error in bash:

> ERROR #85923  GRAPHQL

> There was an error in your GraphQL query:
Cannot query field "images" on type "ContentfulProducts".
If you don't expect "images" to exist on the type "ContentfulProducts" it is most likely a typo.
However, if you expect "images" to exist there are a couple of solutions to common problems: - If you added a new data source and/or changed something inside gatsby-node.js/gatsby-config.js, please try a restart of your development server  - The field might be accessible in another subfield, please try your query in GraphiQL and use the GraphiQL explorer to see which fields you can query and what shape they have - You want to optionally use your field "images" and right now it is not used anywhere. Therefore Gatsby can't infer the type and add it to the GraphQL  schema. A quick fix is to add at least one entry with that field ("dummy content") It is recommended to explicitly type your GraphQL schema if you want to use optional fields. This way you don't have to add the mentioned "dummy content".

Although all "Product" fields in Contentful are exists and therefore all pictures already must be imported with "product" module. Am I right?

**Comment by bagseye** [comment]
Yes, you need to have product images. It requires dummy content to render

**Comment by ibragimof** [comment]
> Yes, you need to have product images. It requires dummy content to render

But I already have product images (which were imported with all demo content from your JSON file in start). Moreover, I see how all my "product" items displaying correctly in website. The only one problem -- pictures in blogs and products are missing.

All photo images lost while importing form Contentful.

**Comment by ibragimof** [comment]
My error about this event (from bash):

> \src\templates\product-template.js
  49:24  warning  'productImages' is assigned a value but never used  no-unused-vars

**Comment by bagseye** [comment]
@ibragimof - did you resolve your issues?

**Comment by ibragimof** [comment]
Yes. Thanks.
Updated with your last commits.
It works for me now.

**Comment by ibragimof** [comment]
I've another question:
Contentful "product" item has field named "faq" (JSON content)
![Untitled](https://user-images.githubusercontent.com/47029060/147088590-f7032b74-c815-4195-832c-02fc20d3113f.png)

and this field responses for creating an "accordion" in product pages.
![nahw](https://user-images.githubusercontent.com/47029060/147088975-78f8a340-4310-43fe-97fc-c56c063f800b.png)


Now I can't format this text inside of "FAQ" boxex. (in last picture it is an orange box). I want to start some lines from new line, create an active links etc.

In JSON field I already try to write "\n\r", markdown format etc... But it didn't works
Is any solution here?

**Comment by bagseye** [comment]
@ibragimof - Yea, that won't work. The implementation is very basic and was first put in around 2 years ago while I was still learning both Gatsby and Contentful. 

I'm in the process of a major overhaul of this theme, creating new content models and changing the way it handles FAQs. The ideal implementation is a repeater field for FAQs but I've not started on it yet. 

My suggestion in the meantime would be to look at creating your own Repeater field implementation with Contentful that replaces the JSON object approach.

**Comment by ibragimof** [comment]
I'll wait your new minor update )
